app/emoji.go

```diff
@@ -187,21 +224,52 @@ func (a *App) GetEmojiByName(emojiName string) (*model.Emoji, *model.AppError) {
 		return nil, model.NewAppError("GetEmoji", "api.emoji.storage.app_error", nil, "", http.StatusNotImplemented)
 	}
 
-	return a.Srv().Store.Emoji().GetByName(emojiName, true)
+	emoji, err := a.Srv().Store.Emoji().GetByName(emojiName, true)
+	if err != nil {
+		var nfErr *store.ErrNotFound
+		var appErr *model.AppError
+		switch {
+		case errors.As(err, &nfErr):
+			return emoji, model.NewAppError("GetEmoji", "store.sql_emoji.get.app_error", nil, err.Error(), http.StatusNotFound)
+		case errors.As(err, &appErr):
+			return emoji, appErr
+		default:
+			return emoji, model.NewAppError("GetEmoji", "store.sql_emoji.get.app_error", nil, err.Error(), http.StatusInternalServerError)
+		}
+	}
+
+	return emoji, nil
 }
 
 func (a *App) GetMultipleEmojiByName(names []string) ([]*model.Emoji, *model.AppError) {
 	if !*a.Config().ServiceSettings.EnableCustomEmoji {
 		return nil, model.NewAppError("GetMultipleEmojiByName", "api.emoji.disabled.app_error", nil, "", http.StatusNotImplemented)
 	}
 
-	return a.Srv().Store.Emoji().GetMultipleByName(names)
+	emoji, err := a.Srv().Store.Emoji().GetMultipleByName(names)
+	if err != nil {
+		switch {
```

Comment:
There's not much use of a switch statement with just a default case. Let's just directly return the apperror.